app/controllers/rb_taskboards_controller.rb

```diff
@@ -41,6 +41,10 @@ class RbTaskboardsController < RbApplicationController
   def show
     @statuses     = Type.find(Task.type).statuses
     @story_ids    = @sprint.stories(@project).map(&:id)
+    if params[:show_sub] and params[:show_sub].to_i == 1
+      @subprojects  = Project.active.where(parent_id: @project.id)
+      @subprojects.each{|p| @story_ids.concat(@sprint.stories(p).map(&:id)) }
+    end
```

Comment:
This does not what you want it to do as it only fetches the immediate children of the project while you want to have all descendants. To do that, you can call:
```
@subprojects = @project.descendants
```

For security reasons, I'd also add another scope to that query which reduces the projects to those visible to the user. This will result in:
```
@subprojects = @project.descendants.visible(current_user)
```

That would fix the immediate problem here. But it leaves another problem which you will not be able to fix here alone. The problem is one of performance. The code will issue n+1 queries which are visible in the console as:

```
Story Load (0.7ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 107 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.5ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 156 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 163 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 168 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 183 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 184 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 203 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 204 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 207 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 208 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 209 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 212 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 213 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 216 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
  Story Load (0.3ms)  SELECT `work_packages`.* FROM `work_packages` WHERE (project_id = 37520 AND type_id in (7,2,8,1,9,10,11,12) AND fixed_version_id in (54))  ORDER BY CASE WHEN work_packages.position IS NULL THEN 1 ELSE 0 END ASC, CASE WHEN work_packages.position IS NULL THEN work_packages.id ELSE work_packages.position END ASC
```

You could alter the code to something like this:

```
@projects = [@project]
@projects << @project.descendants.visible(current_user) if params[:show_sub] and params[:show_sub].to_i == 1 
@story_ids = @sprint.stories(@projects).map(&:id) # maybe even use pluck(:id) instead of map
```

In order for that to work we would have to adapt the implementation of [Story.condition](https://github.com/finnlabs/openproject-backlogs/blob/dev/app/models/story.rb#L178) from
```
c = ['project_id = ? AND type_id in (?) AND fixed_version_id in (?)',
         project_id, Story.types, sprint_ids]
```

to
```
c = ['project_id in (?) AND type_id in (?) AND fixed_version_id in (?)',
         project_id, Story.types, sprint_ids]
```

and maybe rename the `project_id` parameter to `project_ids` for clarity. 

By that you have removed the performance problems I described. Additionally, it would render the changes you have in https://github.com/finnlabs/openproject-backlogs/pull/233/files#diff-8f682c21049311989a264bc4253eba95R137 superfluous because you don't care for the project a story stems from anyway.